server/handlers/policy_relationship_handler.go

```diff
@@ -0,0 +1,128 @@
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/layer5io/meshery/server/models"
+
+	"github.com/ghodss/yaml"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/sirupsen/logrus"
+)
+
+// Define a simple policy struct
+type Policy struct {
+	Rules string `json:"rules"`
+}
+
+type RelationObject struct {
+	DestinationID   string                 `json:"destination_id,omitempty"`
+	DestinationName string                 `json:"destination_name,omitempty"`
+	SourceId        string                 `json:"source_id,omitempty"`
+	SourceName      string                 `json:"source_name,omitempty"`
+	Port            map[string]interface{} `json:"port,omitempty"`
+}
+
+type NetworkPolicyRegoResponse struct {
+	ServicePodRelationships        []RelationObject `json:"service_pod_relationships,omitempty"`
+	ServiceDeploymentRelationships []RelationObject `json:"service_deployment_relationships,omitempty"`
+}
+
+// PolicyRelationshipRegoHandler gets the designFile in the bytes of Yaml and resolves the network relationship as of now.
+func PolicyRelationshipRegoHandler(ctx context.Context, designFile []byte) (*NetworkPolicyRegoResponse, error) {
+	// Load the policy
+	policyFile, err := ioutil.ReadFile("../meshmodel/policies/network-policy.rego") // absolute path, needs to be changed
+	if err != nil {
+		logrus.Fatal("error reading rego file", err.Error())
+		return nil, err
+	}
+
+	// Initialize the policy
+	policy := Policy{string(policyFile)}
+
+	// Initialize the rego engine with the policy
+	engine, err := rego.New(
+		rego.Query("data.network_policy"),
+		rego.Module("policy", policy.Rules),
+	).PrepareForEval(ctx)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var input map[string]interface{}
+	err = yaml.Unmarshal(designFile, &input)
+	if err != nil {
+		return nil, err
+	}
+
+	eval_reponse, err := engine.Eval(ctx, rego.EvalInput(input))
+	if err != nil {
+		return nil, err
+	}
+
+	// Check the result of the policy
+	if len(eval_reponse) > 0 && len(eval_reponse[0].Expressions) > 0 {
+		if eval_resp, ok := (eval_reponse[0].Expressions[0].Value).(map[string]interface{}); ok {
+			return NewRelationPolicy(eval_resp)
+		}
+	}
+
+	return nil, fmt.Errorf("failed to evaluate rego policy %s", eval_reponse)
+}
+
+func (h *Handler) HandleNetworkRelationship(
```

Comment:
Please add swagger comments on top and maybe change the name of the handler to `GetPolicyForDesign` or something similar.

Reply:
Any command to generate swagger comment on the handler func, I'm pretty cautious with adding it by hand. These commands didn't added the comment, but definitely updated the swagger yaml(didn't push there because it will pollute the changes for reviewers to view.)
![Screenshot 2023-05-10 at 3 59 27 AM](https://github.com/meshery/meshery/assets/48255244/2421fdd0-a90a-4192-b77d-8d0c3eb24923)